src/main/java/io/github/ryotan/code/util/CodeEnumReflectionUtil.java

```diff
@@ -0,0 +1,290 @@
+                .filter(CodeEnumReflectionUtil::isCodeEnumType).anyMatch(type -> hasSameParameterizedType(type, aClass));
+    }
+
+    /**
+     * {@link CodeEnum}の実装クラスを返却します。
+     * <p>
+     * {@code code}がEnum型でない場合は、{@link IllegalArgumentException}を送出します。
+     * </p>
+     * 
+     * @param code コード値を表すクラス
+     * @return {@code CodeEnum}の実装クラス
+     * 
+     * @throws IllegalArgumentException {@code code}がEnum型でない場合
+     */
+    @SuppressWarnings("unchecked")
+    public static <C extends CodeEnum<C>> Class<C> getCodeEnumClass(Class<?> code) {
+        if (!isValidCodeEnumClass(code)) {
+            throw new IllegalArgumentException(String.format("%s is not a valid CodeEnum class. "
+                    + "CodeEnum class must be enum and implement CodeEnum<SELF_TYPE>.", code.getName()));
+        }
+        return (Class<C>) code;
+    }
+
+    /**
+     * {@code code}クラスの{@link CodeEnum}で、フィールドやメソッドに指定されたフィルター名が存在した場合、そのオブジェクトを返却します。
+     * <p>
+     * {@code filter}が{@link CodeEnum}のクラスに存在ない場合、{@link IllegalArgumentException}を送出します。
+     * </p>
+     * 
+     * @param code 対象の{@link CodeEnum}のクラス
+     * @param filter   対象の{@link CodeEnum}をフィルタリングするフィルター名
+     * @return 指定されたフィルター名のコードパターン
+     * 
+     * @throws IllegalArgumentException {@code filter}が{@link CodeEnum}のクラスに存在ない場合
+     */
+    public static <C extends CodeEnum<C>> Predicate<C> getCodeFilter(Class<C> code, String filter) {
+        return findCodePatternsFromField(code, filter).orElseGet(() -> findCodePatternsFromMethod(code, filter)
+                .orElseThrow(() -> new IllegalArgumentException(String.format("Code filter '%s' for %s is not found.", filter, code))));
+    }
+
+    /** 
+     * {@code code}クラスで{@link Filter}アノテーションが付与されたフィールドのうち、{@code name}と一致するフィールドを返却します。
+     * 
+     * @param code 対象の{@link CodeEnum}のクラス
+     * @param name     取得した{@link CodeEnum}がもつフィールド名
+     * @return フィールドオブジェクト {@link Predicate}
+     *  
+     */
+    @SuppressWarnings("unchecked")
+    private static <C extends CodeEnum<C>> Optional<Predicate<C>> findCodePatternsFromField(Class<C> code, String name) {
+        try {
+            Field field = code.getField(name);
+            if (isTarget(field, Filter.class, Predicate.class) && isValidCodeFilter(field.getGenericType(), code)) {
+                return Optional.ofNullable((Predicate<C>) field.get(code));
+            }
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            // nop
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * {@code code}クラスで{@link Filter}アノテーションが付与されたメソッドのうち、{@code name}と一致するメソッド名を返却します。
+     * 
+     * @param code 対象の{@link CodeEnum}のクラス
+     * @param name     取得した{@link CodeEnum}がもつメソッド名
+     * @return コードパターン
+     */
+    @SuppressWarnings("unchecked")
+   private static <C extends CodeEnum<C>> Optional<Predicate<C>> findCodePatternsFromMethod(Class<C> code, String name) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected @param tag for 'code'.](https://www.codacy.com/app/ryotan/code-enum-support/file/2976069083/issues/source?bid=3375378&fileBranchId=3375797#l125)